theano/sandbox/rng_mrg.py

```diff
@@ -879,7 +880,7 @@ def gpu_kernels(self, node, name):
             x22 = state_data[idx*6+4];
             x23 = state_data[idx*6+5];
 
-            for (ga_uint i = idx; i < Nsamples; i += Nstreams_used)
+            for (ga_uint i = idx; i < Nsamples; i += step)
```

Comment:
I don't think this is good. We don't re-read the stream seed if we change seed. This is what I think happen if step != Nstreams_used.

I do not think the old back-end need a fix with a rapid check. The new and old back-end don't schedule gs and ls the same way.